comm/src/test/scala/coop/rchain/comm/transport/TransportLayerRuntime.scala

```diff
@@ -58,21 +60,22 @@ abstract class TransportLayerRuntime[F[_]: Monad, E <: Environment] {
   ) extends Runtime[A] {
     def execute(transportLayer: TransportLayer[F], local: PeerNode, remote: PeerNode): F[A]
 
-    def run(): TwoNodesResult =
+    def run(blockUntilDispatched: Boolean = true): TwoNodesResult =
       extract(
         twoNodesEnvironment { (e1, e2) =>
           for {
             localTl  <- createTransportLayer(e1)
             remoteTl <- createTransportLayer(e2)
             local    = e1.peer
             remote   = e2.peer
+            cb       <- createDispatcherCallback
             _ <- remoteTl.receive(
-                  protocolDispatcher.dispatch(remote),
-                  streamDispatcher.dispatch(remote)
+                  protocolDispatcher.dispatch(remote, cb),
+                  streamDispatcher.dispatch(remote, cb)
                 )
             r <- execute(localTl, local, remote)
-            // arbitrary sleep value, so environment has time to handle requests
-            _ <- time.sleep(200 millisecond)
+            _ <- if (blockUntilDispatched) cb.waitUntilDispatched()
+                else implicitly[Timer[F]].sleep(1.second)
```

Comment:
Is there a difference between the syntax `implicitly[Timer[F]].sleep` and `Timer[F].sleep`?

Reply:
@odeac Unfortunately there is no `Timer.apply[F]`